src/models/StyledComponent.js

```diff
@@ -271,6 +293,7 @@ export default (ComponentStyle: Function) => {
 
       const newOptions = {
```

Comment:
I think we might not need this anymore, i.e. we can avoid `options` altogether.

- `attrs` is called before the StyledComponent is created and never again
- `extend` will go
- `withComponent` will go

So any "cloning" is gone

Reply:
Let's wait to refactor this out for v5 when withComponent is gone?